### PR TITLE
make: replace curly braces with parenthesis

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is a follow-up of #8815 but applied to all Makefiles. Then the use of `$(variable)` is consistent across all makefiles (instead of mixing it with `${variable}`).

Maybe @cladmi is interested ?

### CODE REVIEW

**Comment by gebart** [comment]
Did you use a script to perform the replaces?

File: makefiles/buildtests.inc.mk [hunk 1]
```diff
@@ -7,17 +7,17 @@ buildtest:
 	@ \
 	RESULT=true ; \
 	for board in $(BOARDS); do \
-		${COLOR_ECHO} -n "Building for $$board ... " ; \
-		BOARD=$${board} RIOT_CI_BUILD=1 RIOT_VERSION_OVERRIDE=buildtest \
+		$(COLOR_ECHO) -n "Building for $$board ... " ; \
+		BOARD=$$(board) RIOT_CI_BUILD=1 RIOT_VERSION_OVERRIDE=buildtest \
```
Comment (gebart) [inline makefiles/buildtests.inc.mk]: Here, the curly braces matter

**Comment by aabadie** [comment]
> Did you use a script to perform the replaces?

No, just query replaces by hand.

File: makefiles/buildtests.inc.mk [hunk 2]
```diff
@@ -7,17 +7,17 @@ buildtest:
...
+			$(COLOR_ECHO) "$(COLOR_GREEN)success.$(COLOR_RESET)" ; \
 		else \
-			${COLOR_ECHO} "$(COLOR_RED)failed!${COLOR_RESET}" ; \
+			$(COLOR_ECHO) "$(COLOR_RED)failed!$(COLOR_RESET)" ; \
 			RESULT=false ; \
 		fi ; \
 		$(MAKE) clean-intermediates >/dev/null 2>&1 || true; \
 	done ; \
-	$${RESULT}
+	$$(RESULT)
```
Comment (gebart) [inline makefiles/buildtests.inc.mk]: curly braces matter

File: makefiles/info-global.inc.mk [hunk 3]
```diff
@@ -47,36 +47,36 @@ BOARDS := $(filter-out $(BOARDS_WITH_MISSING_FEATURES), $(BOARDS))
 info-buildsizes: SHELL=bash
 info-buildsizes:
 	@echo -e "   text\t   data\t    bss\t    dec\tboard"; \
-	for board in ${BOARDS}; do \
-	    echo "$$(BOARD=$${board} $(MAKE) --no-print-directory info-buildsize 2>/dev/null | tail -n-1 | cut -f-4)" "$${board}"; \
+	for board in $(BOARDS); do \
+	    echo "$$(BOARD=$$(board) $(MAKE) --no-print-directory info-buildsize 2>/dev/null | tail -n-1 | cut -f-4)" "$$(board)"; \
```
Comment (gebart) [inline makefiles/info-global.inc.mk]: curly braces

File: makefiles/info-global.inc.mk [hunk 4]
```diff
@@ -47,36 +47,36 @@ BOARDS := $(filter-out $(BOARDS_WITH_MISSING_FEATURES), $(BOARDS))
...
 	done;
 
 info-buildsizes-diff: SHELL=bash
 info-buildsizes-diff:
 	@echo -e "text\tdata\tbss\tdec\tBOARD/BINDIRBASE\n"; \
-	for board in ${BOARDS}; do \
-	  for BINDIRBASE in $${OLDBIN} $${NEWBIN}; do \
-	      BINDIRBASE=$${BINDIRBASE} BOARD=$${board} $(MAKE) info-buildsize --no-print-directory 2>/dev/null | tail -n-1 | cut -f-4; \
+	for board in $(BOARDS); do \
+	  for BINDIRBASE in $$(OLDBIN) $$(NEWBIN); do \
```
Comment (gebart) [inline makefiles/info-global.inc.mk]: must be curly

File: makefiles/info-global.inc.mk [hunk 5]
```diff
@@ -47,36 +47,36 @@ BOARDS := $(filter-out $(BOARDS_WITH_MISSING_FEATURES), $(BOARDS))
...
+	      BINDIRBASE=$$(BINDIRBASE) BOARD=$$(board) $(MAKE) info-buildsize --no-print-directory 2>/dev/null | tail -n-1 | cut -f-4; \
```
Comment (gebart) [inline makefiles/info-global.inc.mk]: more curlys

File: makefiles/info-global.inc.mk [hunk 6]
```diff
@@ -47,36 +47,36 @@ BOARDS := $(filter-out $(BOARDS_WITH_MISSING_FEATURES), $(BOARDS))
...
 	      fi; \
-	      echo -ne "$${DIFF}\t${COLOR_RESET}"; \
+	      echo -ne "$$(DIFF)\t$(COLOR_RESET)"; \
 	    done; \
-	    echo "$${board}"; \
-	    for I in 0 1 2 3; do echo -ne "$${OLD[I]-${COLOR_RED}ERR${COLOR_RESET}}\t"; done; echo -e "$${OLDBIN}"; \
-	    for I in 0 1 2 3; do echo -ne "$${NEW[I]-${COLOR_RED}ERR${COLOR_RESET}}\t"; done; echo -e "$${NEWBIN}\n"; \
+	    echo "$$(board)"; \
+	    for I in 0 1 2 3; do echo -ne "$$(OLD[I]-$(COLOR_RED)ERR$(COLOR_RESET))\t"; done; echo -e "$$(OLDBIN)"; \
+	    for I in 0 1 2 3; do echo -ne "$$(NEW[I]-$(COLOR_RED)ERR$(COLOR_RESET))\t"; done; echo -e "$$(NEWBIN)\n"; \
```
Comment (gebart) [inline makefiles/info-global.inc.mk]: curly all above

File: makefiles/info-global.inc.mk [hunk 7]
```diff
@@ -47,36 +47,36 @@ BOARDS := $(filter-out $(BOARDS_WITH_MISSING_FEATURES), $(BOARDS))
...
 	  done; \
 	done;
 
 info-boards-supported:
 	@echo $(BOARDS)
 
 info-boards-features-missing:
-	@for f in $(BOARDS_FEATURES_MISSING); do echo $${f}; done | column -t
+	@for f in $(BOARDS_FEATURES_MISSING); do echo $$(f); done | column -t
```
Comment (gebart) [inline makefiles/info-global.inc.mk]: curly

File: makefiles/info.inc.mk [hunk 8]
```diff
@@ -15,7 +15,7 @@ info-objsize:
 	$(SIZE) -d -B $(BASELIBS) | \
 	  tail -n+2 | \
 	  sed -e 's#$(BINDIR)##' | \
-	  sort -rnk$${SORTROW}
+	  sort -rnk$$(SORTROW)
```
Comment (gebart) [inline makefiles/info.inc.mk]: curly

**Comment by gebart** [comment]
I did not mark all mistakes, but look for double dollar signs in the diff and verify those one more time

**Comment by aabadie** [comment]
oups, you are right, I'll fix them.

**Comment by aabadie** [comment]
@gebart, I reverted the wrong changes. I think it should be ok now.

**Comment by aabadie** [comment]
@gebart do you see other problems here ?

**Comment by cladmi** [comment]
I will check this afternoon.

**Pull request merged**